internal/pkg/config/config.go

```diff
@@ -267,3 +332,39 @@ func ReadConfig() (*Config, error) {
 	})
 	return readConfigCached, readConfigCachedErr
 }
+
+func validateFieldNames(bytes *[]byte, matcherType string) error {
+	rawMap := make(map[string]interface{})
+	if err := json.Unmarshal(*bytes, &rawMap); err != nil {
+		return err
+	}
+
+	var validFields []string
+	var validFieldSet map[string]bool
+	switch matcherType {
+	case "fieldTagMatcher":
+		validFields = validFieldTagMatcherFields
+		validFieldSet = validFieldTagMatcherFieldSet
+	case "funcMatcher":
+		validFields = validFuncMatcherFields
+		validFieldSet = validFuncMatcherFieldSet
+	case "sourceMatcher":
+		validFields = validSourceMatcherFields
+		validFieldSet = validSourceMatcherFieldSet
+	}
```

Comment:
What is the benefit of figuring out the fields from the `matcherType` in this function, as opposed to passing in the appropriate fields in each of the callers? A matcher should know what its set of valid fields is. If we add a matcher later, I don't think we should need to change this function.

Reply:
I was thinking of not copying arrays or maps in function calls to improve performance. But I guess that's too much complexity.